api/metrics/v1/labels_enforcer.go

```diff
@@ -180,33 +137,35 @@ func enforceRequestQueryLabels(e *injectproxy.Enforcer, w http.ResponseWriter, r
 
 // Adapted from
 // https://github.com/prometheus-community/prom-label-proxy/blob/952266db4e0b8ab66b690501e532eaef33300596/injectproxy/routes.go.
-func enforceQueryValues(e *injectproxy.Enforcer, v url.Values) (values string, foundQuery bool, err error) {
-	// If no values were given or no query is present,
-	// e.g. because the query came in the POST body
-	// but the URL query string was passed, then finish early.
-	if v.Get(queryParam) == "" {
-		return v.Encode(), false, nil
-	}
-
-	expr, err := parser.ParseExpr(v.Get(queryParam))
-	if err != nil {
-		return "", true, fmt.Errorf("parse expr error: %w", err)
+func enforceQueryValues(e *injectproxy.Enforcer, paramName string, requestParams url.Values) (values string, foundQuery bool, err error) {
+	if len(requestParams[paramName]) == 0 {
+		enforcedMatchers, err := e.EnforceMatchers([]*labels.Matcher{})
+		if err != nil {
+			return "", false, fmt.Errorf("enforce matchers error: %w", err)
+		}
```

Comment:
This is a dirty hack to force the introduction of a `match[]` param to add tenancy enforcement even when the param isn't present. This is needed because `match[]` is optional for label names and values, yet we have to add it to avoid leaks.

Would love to find a cleaner way to do it.

Reply:
Sounds good to me. Maybe might be worth adding this as a comment?

Reply:
Done it [f10aa57](https://github.com/observatorium/api/pull/595/commits/f10aa57d6bfe1a5aa0562ec88581f33f54e35458).